### PR TITLE
[Developer Experience] Add node typing

### PR DESCRIPTION
**Pull request by huchenlei**
Upstreams https://github.com/webfiltered/comfy_typing

This PR adds typing support for ComfyUI nodes so that node developers can have type hints from the IDE.

Some other side effects:
- Converted comfy_types.py to a folder python module
- Annotated `CLIPTextEncode` as an example

### CODE REVIEW

**Comment by comfyanonymous** [comment]
Test is failing, should we drop support for python 3.8 or will you fix it?

**Comment by huchenlei** [comment]
> Test is failing, should we drop support for python 3.8 or will you fix it?

Fixed.

File: .github/workflows/test-launch.yml [hunk 1]
```diff
@@ -28,7 +28,7 @@ jobs:
     - name: Start ComfyUI server
       run: |
         python main.py --cpu 2>&1 | tee console_output.log &
-        wait-for-it --service 127.0.0.1:8188 -t 600
+        wait-for-it --service 127.0.0.1:8188 -t 30
```
Comment (huchenlei) [inline .github/workflows/test-launch.yml:31]: Reduce wait time from 10min to 30s to make the test fail faster.

**Comment by huchenlei** [comment]
> Test is failing, should we drop support for python 3.8 or will you fix it?

Fixed

**Comment by huchenlei** [comment]
@comfyanonymous Ping on review.

**Pull request merged**